### DISCUSSION TITLE
Mod behaves differently when exported as a jar

### DISCUSSION THREAD

**Discussion by KutikiPlayz** [FabricMC/fabric-api#2985] [Mod Dev Support]
I'm making a magic mod, and I have a spell that spews falling blocks in the form of fire in the general direction you're looking at

As you can see in this video (this is in the runClient version of the game that IntelliJ makes), it's working just fine

https://user-images.githubusercontent.com/73810550/229020891-735bdbdc-1816-4c26-9770-c9c2a76ec8e7.mp4

However, when I export the jar and use it as an actual mod, it does this

https://user-images.githubusercontent.com/73810550/229021227-c5259d97-c3aa-4164-8924-bde8de70c27c.mp4

All the mods are the exact same (just fabric API and fabric language kotlin)
Any idea why this could be happening?

**Comment by warjort** [comment]
Posting videos is pointless. We can't debug a video. :-) 

You should put your code on github so we can see all the context and if necessary, debug it for ourselves.

**Reply by KutikiPlayz** [reply]
https://github.com/KutikiPlayz/Cyberpunk-Space-Wizards
the relative files are napalm in the spells folder and moditems in the item folder

**Reply by warjort** [reply]
https://github.com/KutikiPlayz/Cyberpunk-Space-Wizards/blob/117ad602ee8acf28192a831d81aeb0df6c9abe04/src/main/kotlin/kutikiplayz/csw/spell/custom/Napalm.kt#L23

This makes no sense.
You are executing a task using the client thread but then checking `!world.isClient` and launching a co-routine.

As discussed on your other thread. Any code that modifies the game state should be running on the server thread

If you want the client to do something on the server or vice-versa you need to use networking messages
https://fabricmc.net/wiki/tutorial:networking

**Reply by KutikiPlayz** [reply]
I removed the `MinecraftClient.getInstance().execute` on that and everywhere else that had that issue, but it's still doing random stuff in the exported mod.
The confusing bit for me is that all the other spells work perfectly fine (take IceShards for example, since that also summons entities).
Is there anything else specific I should be doing to make sure it works properly?
